src/hooks/useUpload.tsx

```diff
@@ -20,9 +24,13 @@ export const useUpload = (uploadLink = '/api/upload') => {
         setLoading(false);
 
         const res = await response.json();
-        setFiles(res);
-    };
+        if (res.succeeded.length > 0) {
+            onSuccess && onSuccess();
+            setFiles(res);
+        }
 
+        res.failed.length > 0 && onFail && onFail(`Failed to load files ${res.failed.join(', ')}: ${res.errorMessage}`);
```

Comment:
that should be an else statement probably

Reply:
i think it's possible for some files to upload and some to fail in one operation